File: services/device/src/main/java/com/example/iter/device/domain/repository/EquipmentOccupancyRepository.java
package com.example.iter.device.domain.repository;

import com.example.iter.device.domain.entity.EquipmentOccupancy;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;

public interface EquipmentOccupancyRepository extends JpaRepository<EquipmentOccupancy, Long> {

    void deleteByRentalId(Long rentalId);

    void deleteByRentalIdIn(Collection<Long> rentalIds);
}

File: services/device/src/main/kotlin/com/example/iter/device/domain/repository/EquipmentOccupancyRepository.kt
package com.example.iter.device.domain.repository

import com.example.iter.device.domain.entity.EquipmentOccupancy
import org.springframework.data.jpa.repository.JpaRepository

interface EquipmentOccupancyRepository : JpaRepository<EquipmentOccupancy, Long> {

    fun deleteByRentalId(rentalId: Long)

    fun deleteByRentalIdIn(rentalIds: Collection<Long>)
}
